### ISSUE TITLE
add --wait on space delete

### ISSUE THREAD

**Issue by blairham** [devspace-cloud/devspace#1156]
**Is your feature request related to a problem?**
Yes, the problem we have is if you create a space right after deleting a space it creates a space with a random string at the end and is unusable by the user. 

**Which solution do you suggest?**  
adding a flag '--wait' to the devspace delete space.

**Which alternative solutions exist?**  
giving users more RBAC permissions so that they can check to see if their space is removed before creating a new one

**Additional context**  

/kind feature

**Comment by FabianKramm** [comment]
@blairham thanks for reporting this issue! Since `devspace delete space` is not part of the original devspace, I assume you are referring to the [devspace loft plugin](https://github.com/loft-sh/loft-devspace-plugin)? For future issues, it would be great if you could open the issue there. Regarding this issue, I think we can add the `--wait` flag to the space delete command. One note regarding `devspace create space`: currently if the space you are trying to create already exists or is in state Terminating, `devspace create space` will fail and not create a space with a random string at the end (this was the behaviour in devspace-cloud). Furthermore, even if you delete the space, it will still occur in the commands `kubectl get spaces` or `devspace list spaces` as long as it is Terminating.

**Comment by blairham** [comment]
@FabianKramm fair enough where to post, I wasn't sure if it was devspace or the loft command.

Regarding the delete, we were seeing this happen in thedevspace-cloud as we have a bash script that will delete and then create a new space. We ended up having to put an arbitrary wait, otherwise it created these random namespaces. The users don't have permissions to run `kubectl get spaces` we didn't try the devspace list spaces.

**Comment by blairham** [comment]
@FabianKramm here is what I ended up changing too and it works

```
until ! devspace list spaces | grep $spacename > /dev/null
do
:   $(devspace delete space $spacename --cluster development --silent)
done
```

**Comment by FabianKramm** [comment]
@blairham thanks for providing a workaround! We already implemented a `--wait` flag in loftctl and the devspace plugin, which will be released soon!

**Comment by blairham** [comment]
@FabianKramm I believe you have added this feature.

**Comment by FabianKramm** [comment]
@blairham yes thats true, forgot to close the issue, thanks!
